service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/XpathExtractor.java

```diff
@@ -0,0 +1,344 @@
+    private static final String PARTICIPANT_TEMPLATE = "<Participant2 typeCode=\"PPRF\" contextControlCode=\"OP\">%n"
+        + "           <agentRef classCode=\"AGNT\">%n"
+        + "               <id root=\"%s\"/>%n"
+        + "           </agentRef>%n"
+        + "       </Participant2>";
+    private static final String END_OF_LINKSET_COMPONENT = "</LinkSet>\n</component>";
+
+    private static final XPathService X_PATH_SERVICE = new XPathService();
+
+    public static EncounterTemplateParametersBuilder extractValuesForUncategorizedObservation(String component) {
+        var ehrTemplateArgs = EhrTemplateArgs.builder()
+            .availabilityTime(UNCATEGORISED_OBSERVATION_AVAILABILITY_TIME)
+            .authorTime(UNCATEGORISED_OBSERVATION_AUTHOR_TIME)
+            .effectiveTime(UNCATEGORISED_OBSERVATION_EFFECTIVE_TIME_CENTER)
+            .effectiveTimeBackup(UNCATEGORISED_OBSERVATION_EFFECTIVE_TIME_LOW)
+            .authorAgentRef(UNCATEGORISED_OBSERVATION_AUTHOR_REF)
+            .participant2(UNCATEGORISED_OBSERVATION_SECOND_PARTICIPANT)
+            .build();
+
+        return extractValues(component, ehrTemplateArgs);
+    }
+
+    public static EncounterTemplateParametersBuilder extractValuesForCommentObservation(String component) {
+        var ehrTemplateArgs = EhrTemplateArgs.builder()
+            .availabilityTime(COMMENT_OBSERVATION_AVAILABILITY_TIME)
+            .authorTime(COMMENT_OBSERVATION_AUTHOR_TIME)
+            .effectiveTime(COMMENT_OBSERVATION_EFFECTIVE_TIME)
+            .effectiveTimeBackup(COMMENT_OBSERVATION_EFFECTIVE_TIME)
+            .authorAgentRef(COMMENT_OBSERVATION_AUTHOR_REF)
+            .participant2(COMMENT_OBSERVATION_SECOND_PARTICIPANT)
+            .build();
+
+        return extractValues(component, ehrTemplateArgs);
+    }
+
+    public static EncounterTemplateParametersBuilder extractValuesForImmunization(String component) {
+
+        var ehrTemplateArgs = EhrTemplateArgs.builder()
+            .availabilityTime(IMMUNIZATION_AVAILABILITY_TIME)
+            .authorTime(IMMUNIZATION_AUTHOR_TIME)
+            .effectiveTime(IMMUNIZATION_EFFECTIVE_TIME_CENTER)
+            .effectiveTimeBackup(IMMUNIZATION_EFFECTIVE_TIME_LOW)
+            .authorAgentRef(IMMUNIZATION_AUTHOR_REF)
+            .participant2(IMMUNIZATION_SECOND_PARTICIPANT)
+            .build();
+
+        return extractValues(component, ehrTemplateArgs);
+    }
+
+    public static EncounterTemplateParametersBuilder extractValuesForAllergyIntolerance(String component) {
+
+        var ehrTemplateArgs = EhrTemplateArgs.builder()
+            .availabilityTime(ALLERGY_INTOLERANCE_AVAILABILITY_TIME)
+            .authorTime(ALLERGY_INTOLERANCE_AUTHOR_TIME)
+            .effectiveTime(ALLERGY_INTOLERANCE_TIME_CENTER)
+            .effectiveTimeBackup(ALLERGY_INTOLERANCE_TIME_LOW)
+            .authorAgentRef(ALLERGY_INTOLERANCE_AUTHOR_REF)
+            .participant2(ALLERGY_INTOLERANCE_SECOND_PARTICIPANT)
```

Comment:
the typeCode should be PRF not AUT in participant2 
`From source ObservationStatement/Participant/typeCode='PRF'`